engines/router/missionctl/mission_control_upi.go

```diff
@@ -0,0 +1,102 @@
+package missionctl
+
+import (
+	"context"
+
+	"github.com/caraml-dev/turing/engines/router/missionctl/errors"
+	"github.com/caraml-dev/turing/engines/router/missionctl/fiberapi"
+	"github.com/caraml-dev/turing/engines/router/missionctl/instrumentation/metrics"
+	upiv1 "github.com/caraml-dev/universal-prediction-interface/gen/go/grpc/caraml/upi/v1"
+	"github.com/gojek/fiber"
+	fiberGrpc "github.com/gojek/fiber/grpc"
+	fiberProtocol "github.com/gojek/fiber/protocol"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/proto"
+)
+
+type MissionControlUPI interface {
+	Route(context.Context, fiber.Request) (*upiv1.PredictValuesResponse, *errors.TuringError)
```

Comment:
I think the method should return `fiber.Response` instead of `upiv1.PredictValuesResponse` to make it symmetrical. The `Unmarshal` to `upiv1.PredictValuesResponse` should then be handled by caller.